src/components/CohortBuilder/ParticipantsTableView/index.js

```diff
@@ -54,68 +54,91 @@ const ParticipantsTableView = ({
           );
         }
 
+        const dataTotalCount = data[0] ? data[0].total : 0;
         const isRowSelected = node =>
           allRowsSelected || selectedRows.some(row => row === node.participantId);
 
         const dataWithRowSelection = data[0]
           ? data[0].nodes.map(node => ({ ...node, selected: isRowSelected(node) }))
           : [];
 
-        const selectionSQON = selectedRows.length
-          ? {
-              op: 'and',
-              content: [{ op: 'in', content: { field: 'kf_id', value: selectedRows } }],
-            }
-          : sqon;
-
         return (
-          <Card>
-            <ParticipantsTable
-              sqon={selectionSQON}
-              loading={isLoading}
-              data={dataWithRowSelection}
-              dataTotalCount={data[0] ? data[0].total : 0}
-              downloadName={'participant-table'}
-              onFetchData={({ page, pageSize, sorted }) => {
-                const sorting = sorted
-                  .filter(s => SORTABLE_FIELDS_MAPPING.has(s.id))
-                  .map(s => ({
-                    field: SORTABLE_FIELDS_MAPPING.get(s.id),
-                    order: s.desc ? 'desc' : 'asc',
-                  }));
-                setPageIndex(page);
-                setPageSize(pageSize);
-                setSort(sorting);
-              }}
-              onRowSelected={(checked, row) => {
-                const rowId = row.participantId;
-                if (checked) {
-                  setSelectedRows(s => s.concat(rowId));
-                  return;
-                }
-                setSelectedRows(s => s.filter(row => row !== rowId));
-              }}
-              onAllRowsSelected={checked => {
-                // don't keep individual rows selected when "select all" is checked
-                //  to avoid having them selected after "unselect all"
-                setAllRowsSelected(s => checked);
-                setSelectedRows(s => []);
-              }}
-              onClearSelected={() => {
-                setAllRowsSelected(s => false);
-                setSelectedRows(s => []);
-              }}
-              onRemoveFromCohort={() => {
-                // remove the selected participants from the cohort
-                onRemoveFromCohort(selectedRows);
-                // clear selection
-                setAllRowsSelected(s => false);
-                setSelectedRows(s => []);
-              }}
-              selectedRows={selectedRows}
-              allRowsSelected={allRowsSelected}
-            />
-          </Card>
+          <QueriesResolver
```

Comment:
@jaouad-benassila Could we re-use the already "downloaded" JSON dataset for the current page?